static/src/javascripts/projects/common/modules/video/events.js

```diff
@@ -109,36 +110,50 @@ define([
         bindCustomMediaEvents(eventsMap, player, mediaId, mediaType, true);
     }
 
+    function getGoogleAnalyticsEventAction(mediaEvent) {
+        var action = mediaEvent.mediaType + ' ';
+        if (mediaEvent.isPreroll) {
+            action += 'preroll';
+        } else {
+            action += 'content';
+        }
+        return action;
+    }
+
+    function buildGoogleAnalyticsEvent(mediaEvent, metrics, canonicalUrl) {
+        var category = 'Media';
+        var playerName = 'guardian-videojs';
+        var action = getGoogleAnalyticsEventAction(mediaEvent);
+        var fieldsObject = {
+            eventCategory: category,
+            eventAction: action,
+            eventLabel: canonicalUrl,
```

Comment:
@akash1810 Turns out this doesn't work for audio, as the `audio` tag has neither a `data-canonical-url` nor a `data-embed-url`. Maybe it would be easy to add it. Let's sort it out in a separate PR.